### ISSUE TITLE
A dockerised Nextflow instance does not properly mount files to sibling process docker container 

### ISSUE THREAD

**Issue by richardnatarajanAN** [nextflow-io/nextflow#3285]
## Bug report 

I am currently running a nextflow pipeline, which for external reasons will be commenced from within a docker container. To avoid [the infamous docker in docker situation](https://jpetazzo.github.io/2015/09/03/do-not-use-docker-in-docker-for-ci/), the docker is configured to spawn sibling containers by linking the docker.sock file between the host and NF container as below.  

`-v /var/run/docker.sock:/var/run/docker.sock `

Running a nextflow pipeline results in the following error upon reaching any process set to run within a docker container:

```
Error executing process > 'command_name'

Caused by:
  Process `command_name` terminated with an error exit status (127)

  Command executed:
  XXX

Command exit status:
  127

Command output:
  (empty)

Command error:
  /bin/bash: .command.run: No such file or directory

Work dir:
  /usr/src/work/59/d3fbe6adef966d58f1e90fdbdc8054

Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`
```

Further examination of the associated work folder within the NF docker reveals that the `.command.*` files are present as expected, including `.command.run`. However, upon looking closer at the generated `.command.run`, the docker run command appears to attempt to mount the directory as  

```
nxf_launch() {
    docker run -i --cpus 8.0 -e "NXF_DEBUG=${NXF_DEBUG:=0}" -v /mnt/ext4/home:/mnt/ext4/home -v /usr/src/work/b4/31bbfe50afead36f4fa26d74ba123a:/usr/src/work/b4/31bbfe50afead36f4fa26d74ba123a -v "$PWD":"$PWD" -w "$PWD" --entrypoint /bin/bash --name $NXF_BOXID IMAGE_ID -c "/bin/bash /usr/src/work/b4/31bbfe50afead36f4fa26d74ba123a/.command.run nxf_trace"
}
```

I think this is the root of the bug - assuming that the process docker will be spawned as a child, not a sibling, Nextflow attempts to bind mount the working directory of the nextflow docker container to the newly spawned docker container. When this bind mount is accessed as nextflow tries to begin the process, this bind mount actually points to a location on the host, as the host docker daemon is spawning this VM. Therefore no .command.run is found in this location, and the run aborts.

I understand this is expected behaviour under the assumption of typical docker spawning behaviour, which the sibling docker architecture appears to escape. I hope there is a way to allow nextflow to handle sibling docker containers, either a feature or a workaround, especially given this methodology is the typical suggested approach to avoiding DinD.

### Steps to reproduce the problem

1. Build a docker container with nextflow installed
2. Mount the docker host socket to the docker container, allowing sibling containers to be generated instead of nested child containers 
3. Commence a nextflow run with individually dockerised processes

### Log output 

.command.err: `/bin/bash: /usr/src/work/b4/31bbfe50afead36f4fa26d74ba123a/.command.run: No such file or directory`
.command.log: `/bin/bash: /usr/src/work/b4/31bbfe50afead36f4fa26d74ba123a/.command.run: No such file or directory`
.command.out: empty

### Environment 

* Nextflow version: 
```
N E X T F L O W
      version 22.04.5 build 5708
      created 15-07-2022 16:09 UTC 
      cite doi:10.1038/nbt.3820
      http://nextflow.io
```
* Java version: 
```
openjdk version "11" 2018-09-25
OpenJDK Runtime Environment 18.9 (build 11+28)
OpenJDK 64-Bit Server VM 18.9 (build 11+28, mixed mode)
```
* Operating system: 
Host VM = `Ubuntu 16.04.7 LTS`
NF container: `Ubuntu 20.04.5 LTS`

* Bash version: 
```
GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

### Additional context

(Add any other context about the problem here)

**Comment by pditommaso** [comment]
This is a not supported usage scenario

**Comment by richardnatarajanAN** [comment]
Understood and accepted. For anybody seeking to utilise nextflow in this scenario, I found a workaround by using the containerOptions to perform the following two operations for any dockerised NF process:

1. Manually add bind mounts providing access to the required files in a separate directory
2. Added a custom entrypoint which moved the .command.run files and others to $PWD, before executing .command.run nxf_trace as per usual.

This may not be the most elegant solution but it works reliably across any local processes.
